src/Swarm/TUI/Controller.hs

```diff
@@ -150,16 +151,15 @@ toggleModal s modal = do
 --   the updated REPL history to a @.swarm_history@ file.
 shutdown :: AppState -> EventM Name (Next AppState)
 shutdown s = do
+  let hist = mapMaybe getNewEntry (s ^. uiState . uiReplHistory)
+  liftIO $ (`T.appendFile` T.unlines hist) =<< getSwarmHistoryPath True
```

Comment:
Ah yes, this is much better. This way it won't break next time we change the format of the `REPLHistory`.